controllers/powerpod_controller.go

```diff
@@ -347,7 +351,7 @@ func getNewWorkloadContainerList(nodeContainers []powerv1.Container, podStateCon
 
 	logger.V(5).Info("checking if there are new containers for the workload")
 	for _, container := range nodeContainers {
-		if !isContainerInList(container.Name, podStateContainers, logger) {
+		if !isContainerInList(container.Name, container.Id, podStateContainers, logger) {
```

Comment:
Would be nice to create a new unit test for the Reconcile function that would fail on the original code and pass with your proposed changes.